pipeline-runner/R/seurat-2-load_seurat.R

```diff
@@ -53,53 +53,75 @@ reconstruct_seurat <- function(dataset_fpath) {
 
   # get counts
   tryCatch({
+    SeuratObject::DefaultAssay(user_scdata) <- 'RNA'
     counts <- user_scdata[['RNA']]@counts
     test_user_sparse_mat(counts)
     rns <- row.names(counts)
     check_type_is_safe(rns)
-  },
-  error = function(e) {
-    message(e$message)
-    stop(errors$ERROR_SEURAT_COUNTS, call. = FALSE)
-  })
-
-  gene_annotations <- data.frame(input = rns, name = rns, original_name = rns, row.names = rns)
-  user_scdata@misc$gene_annotations <- gene_annotations
 
-  # add dispersions
-  tryCatch({
-    user_scdata <- add_dispersions(user_scdata, method = "default")
-    dispersions <- user_scdata@misc$gene_dispersion
-    test_user_df(dispersions)
   },
   error = function(e) {
     message(e$message)
-    stop(errors$ERROR_SEURAT_HVFINFO, call. = FALSE)
+    stop(errors$ERROR_SEURAT_COUNTS, call. = FALSE)
   })
 
   # get meta data
   tryCatch({
     metadata <- user_scdata@meta.data
+    metadata$active.ident <- user_scdata@active.ident
     test_user_df(metadata)
   },
   error = function(e) {
     message(e$message)
     stop(errors$ERROR_SEURAT_METADATA, call. = FALSE)
   })
 
-  # check for clusters
-  if (!'seurat_clusters' %in% colnames(metadata))
-    stop(errors$ERROR_SEURAT_CLUSTERS, call. = FALSE)
-
-
-  # reconstruct seurat object
+  # reconstruct Seurat object
   scdata <- SeuratObject::CreateSeuratObject(
     counts,
     meta.data = metadata,
   )
 
-  # add dispersions and gene annotations to new scdata
-  scdata@misc$gene_dispersion <- dispersions
+  # add logcounts
+  tryCatch({
+    logcounts <- user_scdata[['RNA']]@data
+    test_user_sparse_mat(logcounts)
+
+    # shouldn't be raw counts
+    suspect.counts <- max(logcounts) > 100
```

Comment:
Totally agree. My motivation here wasn't about speed but rather that I have had a user with a Seurat object where the `@counts` slot was replaced with logcounts. I didn't want to explicitly block the user from exploring the logcounts and warned them that they should not trust any of the between group differential expression analyses. If I were to test for `@data` and `@counts` being identical for this user, it would be `TRUE` and the logcounts would be double-logged.

I think the heuristic is fairly safe as it would take a count value of `x = 2^100 = 1.3e+30` to get `log2(x) = 100`.

 The Seurat object itself was from the [FlyCellAtlas](https://scope.aertslab.org/#/FlyCellAtlas/FlyCellAtlas%2Fr_fca_biohub_body_10x.loom/gene)
 
 We should probably also, at minimum, warn (or potentially disallow) between group DE if the same condition is detected for Seurat projects.

Reply:
I see. Makes sense. I agree on the warning or disallowing between groups DE.  I suspect trajectory analysis would also be affected by the lack of raw counts.